File: paratikaphp/src/srcproject/ParatikaClient.php
<?php
    require_once('ParatikaService.php');
    require_once('ParatikaUtil.php');
    
    $ParatikaClient = new ParatikaClient();
    $ParatikaClient->sessionToken();
    $ParatikaClient->querySession();
    $ParatikaClient->sale();
    $ParatikaClient->preauth();
    
    class ParatikaClient {
        private $ParatikaUtil;
        private $ParatikaService;
        const REQUEST_URL = "> Request URL: ";
        const REQUEST_DATA = "> Request Data: ";
        const RESPONSE_DATA = "> Response Data: ";
        
        /**
         * Test environment API service URL
         * 
         */
        const URL = "https://test.paratika.com.tr/paratika/api/v2";

        public function __construct() {
            $this->ParatikaUtil = new ParatikaUtil();
            $this->ParatikaService = new ParatikaService();
        }
        
        /**
         * Sample API sessionToken operation to the Paratika API
         *
         */
        public function sessionToken() {
            echo '>> A sample SESSIONTOKEN operation is started.' . PHP_EOL;
            $time_started = round(microtime(true) * 1000);
            $parametersMap = $this->ParatikaUtil->prepareSessionTokenParameters();
            $requestData = $this->ParatikaUtil->convertToRequestData($parametersMap);
            echo self::REQUEST_URL . self::URL . PHP_EOL;
            echo self::REQUEST_DATA . $requestData . PHP_EOL;
            
            $request = curl_init(); // create a new cURL resource
            $this->ParatikaService->getConnection($request, self::URL);
            $post_response = $this->ParatikaService->getResponse($request, $requestData);
            
            echo self::RESPONSE_DATA . $post_response . PHP_EOL;
            $this->ParatikaUtil->checkResponse($post_response, 'SESSIONTOKEN');
            echo '>> ParatikaClient.sessionToken() operation is finished in ' . (round(microtime(true) * 1000) - $time_started) . 'ms.' . PHP_EOL;
            return $post_response;
        }
        
        /**
         * Sample API query session operation to the Paratika API in order to see orderItems and extra parameters
         *
         */
        public function querySession() {
            $doSessionToken = self::sessionToken();
            $sessionTokenAsArray = json_decode($doSessionToken, true);
            $sessionToken = $sessionTokenAsArray['sessionToken'];
            echo '>> A sample QUERYSESSION operation is started.' . PHP_EOL;
            $time_started = round(microtime(true) * 1000);
            $parametersMap = $this->ParatikaUtil->prepareQuerySessionParameters($sessionToken);
            $requestData = $this->ParatikaUtil->convertToRequestData($parametersMap);
            echo self::REQUEST_URL . self::URL . PHP_EOL;
            echo self::REQUEST_DATA . $requestData . PHP_EOL;
            
            $request = curl_init(); // create a new cURL resource
            $this->ParatikaService->getConnection($request, self::URL);
            $post_response = $this->ParatikaService->getResponse($request, $requestData);
            
            echo self::RESPONSE_DATA . $post_response . PHP_EOL;
            $this->ParatikaUtil->checkResponse($post_response, 'QUERYSESSION');
            echo '>> ParatikaClient.querySession() operation is finished in ' . (round(microtime(true) * 1000) - $time_started) . 'ms.' . PHP_EOL;
            return $post_response;
        }
        
        /**
         * Sample API sale operation to the Paratika API
         *
         */
        public function sale() {
            echo '>> A sample SALE operation is started.' . PHP_EOL;
            $action = 'SALE';
            $time_started = round(microtime(true) * 1000);
            $parametersMap = $this->ParatikaUtil->preparePrimaryTrxParameters($action);
            $requestData = $this->ParatikaUtil->convertToRequestData($parametersMap);
            echo self::REQUEST_URL . self::URL . PHP_EOL;
            echo self::REQUEST_DATA . $requestData . PHP_EOL;
            
            $request = curl_init(); // create a new cURL resource
            $this->ParatikaService->getConnection($request, self::URL);
            $post_response = $this->ParatikaService->getResponse($request, $requestData);
            
            echo self::RESPONSE_DATA . $post_response . PHP_EOL;
            $this->ParatikaUtil->checkResponse($post_response, $action);
            echo '>> ParatikaClient.sale() operation is finished in ' . (round(microtime(true) * 1000) - $time_started) . 'ms.' . PHP_EOL;
            return $post_response;
        }
        
        /**
         * Sample API preauth operation to the Paratika API
         *
         */
        public function preauth() {
            echo '>> A sample PREAUTH operation is started.' . PHP_EOL;
            $action = 'PREAUTH';
            $time_started = round(microtime(true) * 1000);
            $parametersMap = $this->ParatikaUtil->preparePrimaryTrxParameters($action);
            $requestData = $this->ParatikaUtil->convertToRequestData($parametersMap);
            echo self::REQUEST_URL . self::URL . PHP_EOL;
            echo self::REQUEST_DATA . $requestData . PHP_EOL;
            
            $request = curl_init(); // create a new cURL resource
            $this->ParatikaService->getConnection($request, self::URL);
            $post_response = $this->ParatikaService->getResponse($request, $requestData);
            
            echo self::RESPONSE_DATA . $post_response . PHP_EOL;
            $this->ParatikaUtil->checkResponse($post_response, $action);
            echo '>> ParatikaClient.preauth() operation is finished in ' . (round(microtime(true) * 1000) - $time_started) . 'ms.' . PHP_EOL;
            return $post_response;
        }
    }
?>

File: paratikajava/src/main/java/com/asee/paratikajava/ParatikaClient.java
package com.asee.paratikajava;

import java.net.URL;
import java.net.URLConnection;
import java.util.HashMap;
import java.util.Map;

import org.apache.commons.lang3.StringUtils;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;


public class ParatikaClient {

	private static final String REQUEST_URL = " > Request URL: ";
	private static final String RESPONSE_DATA = " > Response Data: \n";
	private static final String REQUEST_DATA = " > Request Data: ";
	private static final String ACTION = "ACTION";

	private ParatikaUtil paratikaUtil;
	private ParatikaService paratikaService;

	public ParatikaClient() {
	}

	public ParatikaClient(ParatikaUtil paratikaUtil, ParatikaService paratikaService) {
		super();
		this.paratikaUtil = paratikaUtil;
		this.paratikaService = paratikaService;
	}

	/**
	 * Production environment API service URL 
	 */
	 //private static final String URL =  "https://vpos.paratika.com.tr/paratika/api/v2";
	 
	
	/**
	 * Test environment API service URL
	 */
	private static final String URL = "https://test.paratika.com.tr/paratika/api/v2";

	public static void main(String args[]) {
		ParatikaClient paratikaClient = new ParatikaClient();
		paratikaClient.sessionToken();
		paratikaClient.sale();
		paratikaClient.preauth();
		paratikaClient.querySession();
	}

	/**
	 * Sample API sessionToken operation to the Paratika API
	 * 
	 */
	public JSONObject sessionToken() {
		JSONObject paratikaResponse = null;
		try {
			System.out.println("A sample SESSIONTOKEN operation is started.");
			long started = System.currentTimeMillis();
			URL url = new URL(URL);
			System.out.println(REQUEST_URL + URL);
			Map<String, String> requestParameters = new HashMap<String, String>();
			paratikaUtil.prepareSessionTokenParameters(requestParameters);
			String requestData = paratikaUtil.convertToRequestData(requestParameters);
			System.out.println(REQUEST_DATA + requestData);
			URLConnection connection = paratikaService.getConnection(url);
			String response = paratikaService.getResponse(connection, requestData);
			System.out.println(RESPONSE_DATA + response);
			System.out.println(" > ParatikaClient.sessionToken() operation is finished in " + (System.currentTimeMillis() - started) + " ms.");
			JSONParser parser = new JSONParser();
			paratikaResponse = (JSONObject) parser.parse(response);
			checkResponse(paratikaResponse, response, "SESSIONTOKEN");
		} catch (Exception e) {
			e.printStackTrace();
		}
		return paratikaResponse;
	}
	
	/**
	 * Sample API query session operation to the Paratika API in order to see orderItems and extra parameters
	 * 
	 */
	@SuppressWarnings("unchecked")
	public JSONObject querySession() {
		JSONObject paratikaResponse = null;
		try {
			String action = "QUERYSESSION";
			JSONObject doSessionToken = this.sessionToken();
			String sessionToken = (String) doSessionToken.getOrDefault("sessionToken", "");
			System.out.println("A sample QUERYSESSION operation is started.");
			long started = System.currentTimeMillis();
			URL url = new URL(URL);
			System.out.println(REQUEST_URL + URL);
			Map<String, String> requestParameters = new HashMap<String, String>();
			requestParameters.put(ACTION, action);
			requestParameters.put("SESSIONTOKEN", sessionToken);
			String requestData = paratikaUtil.convertToRequestData(requestParameters);
			System.out.println(REQUEST_DATA + requestData);
			URLConnection connection = paratikaService.getConnection(url);
			String response = paratikaService.getResponse(connection, requestData);
			System.out.println(RESPONSE_DATA + response);
			System.out.println(" > ParatikaClient.querySession() operation is finished in " + (System.currentTimeMillis() - started) + "ms.");
			JSONParser parser = new JSONParser();
			paratikaResponse = (JSONObject) parser.parse(response);
			checkResponse(paratikaResponse, response, action);
		} catch (Exception e) {
			e.printStackTrace();
		}
		return paratikaResponse;
	}
	

	/**
	 * Sample API sale operation to the Paratika API
	 * 
	 */
	public JSONObject sale() {
		JSONObject paratikaResponse = null;
		try {
			String action = "SALE";
			System.out.println("A sample SALE operation is started.");
			long started = System.currentTimeMillis();
			URL url = new URL(URL);
			System.out.println(REQUEST_URL + URL);
			Map<String, String> requestParameters = new HashMap<String, String>();
			requestParameters.put(ACTION, action);
			paratikaUtil.preparePrimaryTrxParameters(requestParameters);
			String requestData = paratikaUtil.convertToRequestData(requestParameters);
			System.out.println(REQUEST_DATA + requestData);
			URLConnection connection = paratikaService.getConnection(url);
			String response = paratikaService.getResponse(connection, requestData);
			System.out.println(RESPONSE_DATA + response);
			System.out.println(" > ParatikaClient.sale() operation is finished in " + (System.currentTimeMillis() - started) + "ms.");
			JSONParser parser = new JSONParser();
			paratikaResponse = (JSONObject) parser.parse(response);
			checkResponse(paratikaResponse, response, action);
		} catch (Exception e) {
			e.printStackTrace();
		}
		return paratikaResponse;
	}

	/**
	 * Sample API preauth operation to the Paratika API
	 * 
	 */
	public JSONObject preauth() {
		JSONObject paratikaResponse = null;
		try {
			String action = "PREAUTH";
			System.out.println("A sample PREAUTH operation is started.");
			long started = System.currentTimeMillis();
			URL url = new URL(URL);
			System.out.println(REQUEST_URL + URL);
			Map<String, String> requestParameters = new HashMap<String, String>();
			requestParameters.put(ACTION, action);
			paratikaUtil.preparePrimaryTrxParameters(requestParameters);
			String requestData = paratikaUtil.convertToRequestData(requestParameters);
			System.out.println(REQUEST_DATA + requestData);
			URLConnection connection = paratikaService.getConnection(url);
			String response = paratikaService.getResponse(connection, requestData);
			System.out.println(RESPONSE_DATA + response);
			System.out.println(" > ParatikaClient.preauth() operation is finished in " + (System.currentTimeMillis() - started) + "ms.");
			JSONParser parser = new JSONParser();
			paratikaResponse = (JSONObject) parser.parse(response);
			checkResponse(paratikaResponse, response, action);
		} catch (Exception e) {
			e.printStackTrace();
		}
		return paratikaResponse;
	}

	private void checkResponse(JSONObject json, String response, String action) {
		if (response != null
				&& StringUtils.equalsIgnoreCase(json.get("responseCode").toString(), "00")
				&& StringUtils.equalsIgnoreCase(json.get("responseMsg").toString(), "Approved")) {
			System.out.println(" > " + action + " operation is SUCCESSFUL.");
		} else {
			System.out.println(" > " + action + " operation is FAILED, please check the error codes.");
			System.out.println("   ErrorCode     : " + json.get("errorCode"));
			System.out.println("   ErrorMessage  : " + json.get("errorMsg"));
		}
	}
}
